packages/service-provider-core/src/printable-bson.ts

```diff
@@ -43,10 +44,38 @@ export const bsonStringifiers: Record<
   BSONRegExp: makeClasslessInspect('BSONRegExp'),
   Binary: function (
     this: typeof BSON.Binary.prototype,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
     ...args: any[]
   ): string {
     const hexString = this.toString('hex');
     switch (this.sub_type) {
+      case BSON.Binary.SUBTYPE_VECTOR:
+        switch (this.buffer[0]) {
+          case BSON.Binary.VECTOR_TYPE.Int8:
+            return `Int8(${utilInspect(
+              Array.from(this.toInt8Array()),
+              // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+              ...args
+            )})`;
+          case BSON.Binary.VECTOR_TYPE.Float32:
+            return `Float32(${utilInspect(
+              Array.from(this.toFloat32Array()),
```

Comment:
not sure if this is a problematically heavy operation here, `utilInspect`-ing directly causes it to display like a `{'0': 1... }` so not really what we want.

Reply:
This is what I get without the `Array.from()` bit:

```js
test> db.bsonvec.findOne()
{
  _id: ObjectId('67ed08d64a05814b92a9ff62'),
  x: Float32(Float32Array(1) [ 1.539989614439558e-36 ])
}
```

Reply:
Ah, I was testing in it in a playground which was using the DOM implementation for util... that makes more sense